src/Server.cpp

```diff
@@ -656,6 +626,12 @@ void cServer::KickUser(int a_ClientID, const AString & a_Reason)
 
 void cServer::AuthenticateUser(int a_ClientID, const AString & a_Name, const AString & a_UUID, const Json::Value & a_Properties)
 {
+	if (GetNumPlayers() >= GetMaxPlayers())
```

Comment:
As long the comparison is done on the result of the `fetch_add` then It shouldn't be possible.